clients/fides-js/src/components/notices/NoticeOverlay.tsx

```diff
@@ -24,17 +21,29 @@ import { useConsentServed } from "../../lib/hooks";
 import { updateCookieFromNoticePreferences } from "../../lib/cookie";
 import PrivacyPolicyLink from "../PrivacyPolicyLink";
 import { dispatchFidesEvent } from "../../lib/events";
+import { resolveConsentValue } from "../../lib/consent-value";
+import { getConsentContext } from "../../lib/consent-context";
+import { transformConsentToFidesUserPreference } from "../../lib/shared-consent-utils";
 
 const NoticeOverlay: FunctionComponent<OverlayProps> = ({
   experience,
   options,
   fidesRegionString,
   cookie,
 }) => {
-  const initialEnabledNoticeKeys = useMemo(
-    () => Object.keys(cookie.consent).filter((key) => cookie.consent[key]),
-    [cookie.consent]
-  );
+  const consentContext = useMemo(() => getConsentContext(), []);
+  const initialEnabledNoticeKeys = useMemo(() => {
+    if (experience.privacy_notices) {
+      return experience.privacy_notices?.map((notice) => {
+        const val = resolveConsentValue(notice, consentContext, cookie);
+        if (val) {
+          return notice.notice_key as PrivacyNotice["notice_key"];
+        }
+        return "";
```

Comment:
nit: ternary is more concise

```suggestion
        return val ? notice.notice_key as PrivacyNotice["notice_key"] : "";
```

Reply:
is this cast needed because we are going from `PrivacyNoticeWithPreference` to `PrivacyNotice`?

Reply:
yup exactly